kratos/containers/distributed_csr_matrix.h

```diff
@@ -69,17 +69,20 @@ class DistributedCsrMatrix
     ///@}
     ///@name Life Cycle
     ///@{
+    DistributedCsrMatrix() //default constructor. To be used for low level operations
+        : mrComm(ParallelEnvironment::GetDefaultDataCommunicator())
```

Comment:
I would not allow this
this can lead to nasty deadlocks
IMO the user must know which data-comm to use